src/main/java/es/us/isa/restest/generators/FuzzingTestCaseGenerator.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.javatuples.Pair;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.*;
+
+/**
+ * This class implements a generator of fuzzing test cases. It uses a customizable dictionary to obtain
+ * fuzzing parameters. Those parameters are classified by type (string, integer, number, boolean).
+ *
+ * @author José Ramón Fernández
+ */
+
+public class FuzzingTestCaseGenerator extends AbstractTestCaseGenerator {
+
+    private Map<String, List<String>> fuzzingMap;
+
+    private static Logger logger = LogManager.getLogger(FuzzingTestCaseGenerator.class.getName());
+
+    public FuzzingTestCaseGenerator(OpenAPISpecification spec, TestConfigurationObject conf, int nTests) {
+        super(spec, conf, nTests);
+        TypeReference<HashMap<String, List<String>>> typeRef = new TypeReference<HashMap<String, List<String>>>() {
+        };
+        try {
+            this.fuzzingMap = new ObjectMapper().readValue(FileManager.readFile("src/main/resources/fuzzing-dictionary.json"), typeRef);
+        } catch (JsonProcessingException e) {
+            logger.error("Error processing JSON fuzzing dictionary", e);
+        }
+    }
+
+    @Override
+    protected Collection<TestCase> generateOperationTestCases(Operation testOperation) {
+
+        List<TestCase> testCases = new ArrayList<>();
+
+        resetOperation();
+
+        while (hasNext()) {
+            TestCase test = generateNextTestCase(testOperation);
+            test.setFulfillsDependencies(false);
+            test.setFaulty(false);
+
+            authenticateTestCase(test);
+            testCases.add(test);
+            updateIndexes(test);
+        }
+
+        return testCases;
+    }
+
+    @Override
+    public TestCase generateNextTestCase(Operation testOperation) {
+        TestCase tc = createTestCaseTemplate(testOperation);
+
+        if (testOperation.getTestParameters() != null) {
+            for (TestParameter testParam : testOperation.getTestParameters()) {
+                if (!testParam.getIn().equals("body")) {
+                    generateFuzzingParameter(tc, testParam, testOperation);
+                } else {
+                    generateFuzzingBody(tc, testParam, testOperation);
+                }
+            }
+        }
+        return tc;
+    }
+
+    private void generateFuzzingParameter(TestCase tc, TestParameter testParam, Operation testOperation) {
+        if (testParam.getWeight() == null || rand.nextFloat() <= testParam.getWeight()) {
+            ParameterFeatures param = SpecificationVisitor.findParameter(testOperation.getOpenApiOperation(), testParam.getName(), testParam.getIn());
+            List<String> fuzzingList = fuzzingMap.get(param.getType());
+            tc.addParameter(testParam, fuzzingList.get(rand.nextInt(fuzzingList.size())));
+        }
+    }
```

Comment:
if the parameter is an enum, the list of enum values should be considered (in addition to the fuzzing values).